### PR TITLE
fix(contentful): properly delete deleted entries and assets

### LINKED ISSUE #5747: [source-contentful] node deletions bug
If you hook up your project with a Contentful space that has deleted blog posts, they are still being fetched by GraphQL. It seems like CF doesn't delete the nodes once they are created--they just get returned with `null` values:

nodes

The issue this presents is in the index page (src/pages/index.js) because the code is looping through each node and creating a `div` for it (so a user could end up with empty white space on their blog if they have deleted entries--I was able to recreate):
```
            {posts.map(({ node }) => {
              return (
                <div className="article" key={node.slug}>
                  <ArticlePreview article={node} />
                </div>
              )
            })}
```

I was looking at this issue with another engineer and the easiest solution we could come up with is to only use nodes that have a valid title. So instead of this:
```
const posts = get(this, 'props.data.allContentfulBlogPost.edges')
```

You can filter for only posts that have a title (or any other field):
```
const allPosts = get(this, 'props.data.allContentfulBlogPost.edges')
const posts = allPosts.filter(post => post.node.title)
```

Question: Is it expected behavior to have phantom nodes even after posts are deleted? And should I create a PR to filter out these nodes if you agree with my assessment? 

@zcei Had a theory
> From what I found, there might a bug here, as [deleted entries are handled by `gatsby-source-contentful`](https://github.com/gatsbyjs/gatsby/blob/ae8e1ccc77f89a13a9828c5c200b6e45e47470c0/packages/gatsby-source-contentful/src/fetch.js#L75-L80).

> I haven't worked a lot with the source, but it looks like we're using the (maybe `c`-prefixed) [entity ID to delete a node](https://github.com/gatsbyjs/gatsby/blob/ae8e1ccc77f89a13a9828c5c200b6e45e47470c0/packages/gatsby-source-contentful/src/gatsby-node.js#L78-L80), while [a different format](https://github.com/gatsbyjs/gatsby/blob/ae8e1ccc77f89a13a9828c5c200b6e45e47470c0/packages/gatsby-source-contentful/src/normalize.js#L70) is used to [create the node](https://github.com/gatsbyjs/gatsby/blob/ae8e1ccc77f89a13a9828c5c200b6e45e47470c0/packages/gatsby-source-contentful/src/normalize.js#L317).

### PR DESCRIPTION
**Pull request by axe312ger**
fixes #5747
 
This should ensure that the correct id's are passed to deleteNode

### CODE REVIEW

File: packages/gatsby-source-contentful/src/gatsby-node.js [hunk 1]
```diff
@@ -75,12 +75,17 @@ exports.sourceNodes = async (
   // Remove deleted entries & assets.
   // TODO figure out if entries referencing now deleted entries/assets
   // are "updated" so will get the now deleted reference removed.
+
+  function deleteContentfulNode (node) {
+    const id = normalize.fixId(node.sys.id)
```
Comment (pieh) [inline packages/gatsby-source-contentful/src/gatsby-node.js]: isn't `normalize.fixId` already done here - https://github.com/axe312ger/gatsby/blob/3058f95dc020524dec1242da3523a1e9a5f73e5d/packages/gatsby-source-contentful/src/fetch.js#L75-L86 (`fixIds` seem to traverse passed object and run `fixId` on `id` fields) ?

Comment (KyleAMathews) [reply packages/gatsby-source-contentful/src/gatsby-node.js]: Also the id is passed to your new function not the node. You need to fix the id and then use that to get the node.

**Comment by pieh** [comment]
Tested it locally and this now properly deletes nodes for deleted entries of non-default locales. But I couldn't replicate exact issue presented in https://github.com/gatsbyjs/gatsby/issues/5747 (nodes with null fields). I would see nodes that should be deleted but containing all data before applying changes from this PR.

File: packages/gatsby-source-contentful/src/gatsby-node.js (hunk 1 above)
Comment (pieh) [reply packages/gatsby-source-contentful/src/gatsby-node.js]: node is passed there - lines that maps node -> id are deleted ( lines 79, 82 ), but I checked again and we can skip normalizing id here (it's already normalized)

**Comment by axe312ger** [comment]
I removed the `normalize.fixID` as suggested.

**Pull request merged**